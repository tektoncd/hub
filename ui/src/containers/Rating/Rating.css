.hub-details-rating {
  width: 20em;
}

.hub-rate-area {
  float: left;
  border-style: none;
}

.hub-rate-area:not(:checked) > input {
  position: absolute;
  top: -9999px;
}

.hub-rate-area:not(:checked) > label {
  float: right;
  width: 1em;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  font-size: 170%;
  line-height: 1.2;
  color: lightgrey;
}

.hub-rate-area > label > svg {
  color: #484848;
}

.hub-rate-area:not(:checked) > label:before {
  content: '';
}

.hub-rate-area > input:checked ~ label {
  color: #484848;
  text-shadow: 1px 1px #484848;
  font-size: 170% !important;
}

.hub-rate-area:not(:checked) > label:hover,
.hub-rate-area:not(:checked) > label:hover ~ label {
  color: #484848;
}

.hub-rate-area > input:checked + label:hover,
.hub-rate-area > input:checked + label:hover ~ label,
.hub-rate-area > input:checked ~ label:hover,
.hub-rate-area > input:checked ~ label:hover ~ label,
.hub-rate-area > label:hover ~ input:checked ~ label {
  color: #484848;
  text-shadow: 1px 1px #484848;
}

.hub-rate-area > label:active {
  position: relative;
  top: 2px;
  left: 2px;
}
