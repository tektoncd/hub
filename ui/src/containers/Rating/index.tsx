import React from 'react';
import { StarIcon } from '@patternfly/react-icons';
import './Rating.css';

const Rating: React.FC = () => {
  return (
    <div className="hub-details-rating">
      <form>
        <ul>
          {Array.apply(0, Array(5)).map((_, element) => (
            <li key={`${element + 1}-input`} className="hub-rate-area">
              <input type="radio" id={`${element}-star`} name="rating" value={`${element}`} />
              <label htmlFor={`${element}-star`}>
                <StarIcon />
              </label>
            </li>
          ))}
        </ul>
      </form>
    </div>
  );
};

export default Rating;
