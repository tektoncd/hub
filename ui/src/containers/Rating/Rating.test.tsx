import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '.';

it('should render Ratings component', () => {
  const component = renderer.create(<Rating />).toJSON();
  expect(component).toMatchSnapshot();
});
