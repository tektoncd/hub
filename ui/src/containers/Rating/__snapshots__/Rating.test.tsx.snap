// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Ratings component 1`] = `
<div
  className="hub-details-rating"
>
  <form>
    <ul>
      <li
        className="hub-rate-area"
      >
        <input
          id="0-star"
          name="rating"
          type="radio"
          value="0"
        />
        <label
          htmlFor="0-star"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              transform=""
            />
          </svg>
        </label>
      </li>
      <li
        className="hub-rate-area"
      >
        <input
          id="1-star"
          name="rating"
          type="radio"
          value="1"
        />
        <label
          htmlFor="1-star"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              transform=""
            />
          </svg>
        </label>
      </li>
      <li
        className="hub-rate-area"
      >
        <input
          id="2-star"
          name="rating"
          type="radio"
          value="2"
        />
        <label
          htmlFor="2-star"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              transform=""
            />
          </svg>
        </label>
      </li>
      <li
        className="hub-rate-area"
      >
        <input
          id="3-star"
          name="rating"
          type="radio"
          value="3"
        />
        <label
          htmlFor="3-star"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              transform=""
            />
          </svg>
        </label>
      </li>
      <li
        className="hub-rate-area"
      >
        <input
          id="4-star"
          name="rating"
          type="radio"
          value="4"
        />
        <label
          htmlFor="4-star"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              transform=""
            />
          </svg>
        </label>
      </li>
    </ul>
  </form>
</div>
`;
